### PR TITLE
rocSOLVER - Changes to support multiple ROCM installation

### PR DESCRIPTION
**Pull request by pruthvistony**
- Adding rocm_set_soversion to set versioning to the lib so file.

Signed-off-by: Pruthvi Madugundu <mpruthvi@gmail.com>

### CODE REVIEW

File: library/src/CMakeLists.txt [hunk 1]
```diff
@@ -95,6 +95,7 @@ target_include_directories( rocsolver
 
 set_target_properties( rocsolver PROPERTIES VERSION ${rocsolver_VERSION} SOVERSION ${rocsolver_SOVERSION} CXX_EXTENSIONS NO )
 set_target_properties( rocsolver PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/staging" )
+rocm_set_soversion(rocsolver ${VERSION_STRING})
```
Comment (jzuniga-amd) [inline library/src/CMakeLists.txt]: Just a little modif here:
what we need is to set the .so version; and if using rocm_set_soversion, let's remove it from the set_target_properies statement, so:

set_target_properties( rocsolver PROPERTIES CXX_EXTENSIONS NO )
set_target_properties( rocsolver PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/staging" )
rocm_set_soversion(rocsolver ${rocsolver_SOVERSION})

Comment (pruthvistony) [reply library/src/CMakeLists.txt]: The variable - rocsolver_SOVERSION value is 0. 
For rocm_set_soversion function need to send major.minor version string.

Comment (jzuniga-amd) [reply library/src/CMakeLists.txt]: Oh, right! It should be 
set( rocsolver_SOVERSION 0.1 )
in library/CMakeLists.txt

File: CMakeLists.txt [hunk 2]
```diff
@@ -136,7 +136,7 @@ get_os_id(OS_ID)
 message (STATUS "OS detected is ${OS_ID}")
 
 # Versioning via rocm-cmake
-set ( VERSION_STRING "2.7.0" )
```
Comment (saadrahim) [inline CMakeLists.txt]: Library versioning should be maintained by project owner @jzuniga-amd

File: library/src/CMakeLists.txt (hunk 1 above)
Comment (saadrahim) [reply library/src/CMakeLists.txt]: SOVERSION should be 0.1 or 1.0 not 2.7. I prefer 1.0 but @jzuniga-amd  should decide

**Comment by saadrahim** [comment]
Implemented in a separate PR

**Pull request closed without merging**